source/Applications/Basic/Visualization/CaptureHDRVisNormals/CaptureHDRVisNormals.cpp

```diff
@@ -0,0 +1,143 @@
+            cloud->points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return cloud;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGBNormal>::Ptr zividToPclVisualizationNormals(
+        const Zivid::Array2D<Zivid::PointXYZColorRGBA> &data,
+        const Zivid::Array2D<Zivid::NormalXYZ> &normals)
+    {
+        auto normalsPcl = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGBNormal>>();
+        normalsPcl->width = data.width();
+        normalsPcl->height = data.height();
+        normalsPcl->is_dense = false;
+        normalsPcl->points.resize(data.size());
+        for(size_t i = 0; i < data.size(); ++i)
+        {
+            normalsPcl->points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            // Scale colors for normals between 0 and 1 to create a normal map
+            normalsPcl->points[i].r = 0.5 * (1 - normals(i).x) * 255; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].g = 0.5 * (1 - normals(i).y) * 255; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].b = 0.5 * (1 - normals(i).z) * 255; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].normal_x = normals(i).x;            // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].normal_y = normals(i).y;            // NOLINT(cppcoreguidelines-pro-type-union-access)
+            normalsPcl->points[i].normal_z = normals(i).z;            // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return normalsPcl;
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring settings" << std::endl;
+        Zivid::Settings settings;
+        for(const auto aperture : { 9.57, 4.76, 2.59 })
+        {
+            std::cout << "Adding acquisition with aperture = " << aperture << std::endl;
+            const auto acquisitionSettings = Zivid::Settings::Acquisition{
+                Zivid::Settings::Acquisition::Aperture{ aperture },
+            };
+            settings.acquisitions().emplaceBack(acquisitionSettings);
+        }
+
+        std::cout << "Capturing frame (HDR)" << std::endl;
+        const auto frame = camera.capture(settings);
+        const auto pointCloud = frame.pointCloud();
+
+        std::cout << "Creating PCL point cloud structure" << std::endl;
+        const auto data = pointCloud.copyData<Zivid::PointXYZColorRGBA>();
+        const auto cloud = zividToPclPoints(data);
```

Comment:
Check how we call this variable in other PCL samples, for consistency. I would ads Pcl to name like inside the function.

Reply:
I checked and we call it pointCloudPCL in other samples so I suggest that and normalsPCL